src/HealthChecks.Solr/SolrHealthCheck.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using SolrNet;
+using SolrNet.Impl;
+using SolrNet.Impl.ResponseParsers;
+using SolrNet.Commands;
+
+namespace HealthChecks.Solr
+{
+    public class SolrHealthCheck
+        : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, SolrConnection> _connections = new ConcurrentDictionary<string, SolrConnection>();
+
+        private readonly SolrOptions _options;
+
+        public SolrHealthCheck(SolrOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var url = $"{_options.Uri}/{_options.Core}";
+
+                if (!_connections.TryGetValue(url, out SolrConnection solrConnection))
+                {
+                    solrConnection = new SolrConnection(url);

```

Comment:
Is `SolrConnection` disposable resource?

Reply:
It's not disposable.